src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/MobDropListener.java

```diff
@@ -48,14 +50,20 @@ public void onEntityKill(EntityDeathEvent e) {
     }
 
     private void addDrops(Player p, Set<ItemStack> customDrops, List<ItemStack> drops) {
+    	int random = ThreadLocalRandom.current().nextInt(100);
+    	
         for (ItemStack drop : customDrops) {
             if (Slimefun.hasUnlocked(p, drop, true)) {
-                if (circuitBoard != null && circuitBoard.isItem(drop) && !circuitBoard.isDroppedFromGolems()) {
+            	SlimefunItem sfi = SlimefunItem.getByItem(drop);
+            	if (sfi instanceof RandomMobDrop && ((RandomMobDrop)sfi).getDropChance() <= random) 
+            		continue;
+            	
+                if (circuitBoard != null && circuitBoard.isItem(drop) && !circuitBoard.isDroppedFromGolems()) 
```

Comment:
No need to even have this CircuitBoard variable at this point anymore. You can just check if `sfi` is an instance of your Circuit Board, would be much better performance wise since you already done these operations earlier.